pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -188,7 +188,9 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 	// always update the elasticsearch state bits
 	d.ReconcileState.UpdateElasticsearchState(*resourcesState, observedState())
 
-	if err := d.verifySupportsExistingPods(resourcesState.CurrentPods); err != nil {
+	_, permissive := d.ES.Annotations[esv1.DisableDowngradeValidationAnnotation]
+
+	if err := d.verifySupportsExistingPods(resourcesState.CurrentPods); !permissive && err != nil {
```

Comment:
If `permissive` is `true` and, if the validation fails, I think I would still log the validation result and mention that the downgrade is allowed as requested by the user. It might help to understand some situations afterwards by reading the logs.

Reply:
Good point, will change it.